File: src/main/java/de/swt/testing/CheckoutService.java
package de.swt.testing;

/**
 * Service that handles the checkout process using a payment gateway.
 */
public class CheckoutService {

    private final PaymentGateway paymentGateway;

    public CheckoutService(PaymentGateway paymentGateway) {
        this.paymentGateway = paymentGateway;
    }

    /**
     * Performs checkout with the given card and amount.
     *
     * @param cardNumber the card number
     * @param amount the payment amount
     * @return "INVALID_AMOUNT" if amount is not positive,
     *         "SUCCESS" if payment succeeds,
     *         "PAYMENT_FAILED" otherwise
     */
    public String checkout(String cardNumber, double amount) {
        if (amount <= 0) {
            return "INVALID_AMOUNT";
        }
        boolean success = paymentGateway.processPayment(cardNumber, amount);
        return success ? "SUCCESS" : "PAYMENT_FAILED";
    }
}

File: python/src/checkout_service.py
"""Service that handles the checkout process using a payment gateway."""


class CheckoutService:
    """Handles the checkout process using a payment gateway.

    The payment_gateway must be any object with a
    ``process_payment(card_number, amount)`` method that returns a bool.
    """

    def __init__(self, payment_gateway):
        self._payment_gateway = payment_gateway

    def checkout(self, card_number, amount):
        """Perform checkout with the given card and amount.

        Args:
            card_number: The card number.
            amount: The payment amount.

        Returns:
            "INVALID_AMOUNT" if amount is not positive,
            "SUCCESS" if payment succeeds,
            "PAYMENT_FAILED" otherwise.
        """
        if amount <= 0:
            return "INVALID_AMOUNT"
        success = self._payment_gateway.process_payment(card_number, amount)
        return "SUCCESS" if success else "PAYMENT_FAILED"
